### PR TITLE
arm64: need to restore the sentry's TLS when in EL1

### PR DESCRIPTION
**Pull request by lemin9538**
when exist to EL1, the TLS in sentry is still use the application's TLS.

### CODE REVIEW

**Comment by avagin** [comment]
Cc: @lubinszARM

**Comment by lubinszARM** [comment]
Hi @lemin9538 
Can you add such a test case for this pr in kvm_test?
Thanks. :)

**Comment by lemin9538** [comment]
> Hi @lemin9538
> Can you add such a test case for this pr in kvm_test?
> Thanks. :)

Ok

File: pkg/sentry/platform/ring0/entry_arm64.s [hunk 1]
```diff
@@ -516,6 +516,10 @@ TEXT ·kernelExitToEl1(SB),NOSPLIT,$0
 	MOVD CPU_REGISTERS+PTRACE_PC(RSV_REG), R1
 	MSR R1, ELR_EL1
 
+	// restore sentry's tls.
+	MOVD CPU_REGISTERS+PTRACE_TLS(RSV_REG), R1
+	MSR R1, TPIDR_EL0
```
Comment (lubinszARM) [inline pkg/sentry/platform/ring0/entry_arm64.s:521]: Agree.

File: pkg/sentry/platform/ring0/entry_arm64.s [hunk 2]
```diff
@@ -540,6 +544,10 @@ TEXT ·Start(SB),NOSPLIT,$0
 	ORR $0xffff000000000000, RSV_REG, RSV_REG
 	WORD $0xd518d092        //MSR R18, TPIDR_EL1
 
+	// save sentry's tls to memory.
+	MRS TPIDR_EL0, R1
+	MOVD R1, CPU_REGISTERS+PTRACE_TLS(RSV_REG)
```
Comment (lubinszARM) [inline pkg/sentry/platform/ring0/entry_arm64.s:549]: In my opinion, sentry's tls has already been saved into memory in bluepillArchEnter().
So, this code is redundant

Comment (lemin9538) [reply pkg/sentry/platform/ring0/entry_arm64.s]: Yes, you are right. bluepillArchEnter() has already save the TLS

**Comment by lubinszARM** [comment]
LGTM

**Pull request merged**